extensions/elytron-security-oauth2/runtime/src/main/java/io/quarkus/elytron/security/oauth2/runtime/OAuth2Config.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.elytron.security.oauth2.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * See http://docs.wildfly.org/14/WildFly_Elytron_Security.html#validating-oauth2-bearer-tokens
+ */
+@ConfigRoot(name = "oauth2", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class OAuth2Config {
+    /**
+     * If the embedded store is enabled.
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean enabled = true;
+
+    /**
+     * The identifier of the client on the OAuth2 Authorization Server
+     */
+    @ConfigItem
+    public String clientId;
+
+    /**
+     * The secret of the client
+     */
+    @ConfigItem
+    public String clientSecret;
+
+    /**
+     * The URL of token introspection endpoint
+     */
+    @ConfigItem
+    public String introspectionUrl;
+
+    /**
+     * The path to a ca custom cert file
+     */
+    @ConfigItem
+    public Optional<String> caCertFile;
+
+    /**
+     * The claim that provides the roles
+     */
+    @ConfigItem(defaultValue = "scope")
+    public String roleClaim;
+
+    /**
```

Comment:
@loicmathieu I was creating a smallrye-jwt issue to correctly handle 'scope' (thanks for reminding me about the way it is structured) and I decided against introducing a property similar to `roleSeparator` which you introduce here.
I think it would be better, for the moment, remove it and hard code the `scope` check in this extensions for now. Ideally the configuration for processing the custom roles claim would be atomic; if one has a claim like `customrole` which is just a simple value then the expectation that if the `roleSeparator` is empty then the `customrole` is an array does not hold. 
Later, once @pedroigor PR is merged, one can provide a custom roles decoder and push all this code there.
@loicmathieu @pedroigor would you agree ?

Reply:
@sberyozkin OK for removing the scope delimiter, I will do the following:
- If the claim is a string : split by space (it's in the spec that the introspection endpoint returns scopes separated by a space)
- If it's a List<String> : return the list.

Later we can introduce a RoleDecoder (that will use the claims) and offers a way to the user to provide it's own. I will open a PR for this after this one will be merged.

Reply:
OK for me

Reply:
@loicmathieu sure, I was only suggesting to continue doing a `scope` specific processing internally and clean it up later as opposed to generalizing right now with the `roleSeparator` but at the cost of having some minor ambiguities as a result